crates/smithy-core/src/generator.rs

```diff
@@ -0,0 +1,296 @@
+                .entry(model.namespace.clone())
+                .or_default()
+                .push(model);
+        }
+
+        for (namespace, models) in namespace_models {
+            let filename = format!("{}.smithy", namespace.replace('.', "_"));
+            let filepath = output_dir.join(filename);
+
+            let mut content = String::new();
+            content.push_str("$version: \"2\"\n\n");
+            content.push_str(&format!("namespace {}\n\n", namespace));
+
+            // Collect all unique shape definitions for the current namespace
+            let mut shapes_in_namespace = HashMap::new();
+            for model in models {
+                for (shape_name, shape) in &model.shapes {
+                    shapes_in_namespace.insert(shape_name.clone(), shape.clone());
+                }
+            }
+
+            // Generate definitions for each shape in the namespace
+            for (shape_name, shape) in &shapes_in_namespace {
+                content.push_str(&self.generate_shape_definition(
+                    shape_name,
+                    shape,
+                    &namespace,
+                    &shape_to_namespace,
+                ));
+                content.push_str("\n\n");
+            }
+
+            fs::write(filepath, content)?;
+        }
+
+        Ok(())
+    }
+
+    fn generate_shape_definition(
+        &self,
+        name: &str,
+        shape: &crate::types::SmithyShape,
+        current_namespace: &str,
+        shape_to_namespace: &HashMap<String, String>,
+    ) -> String {
+        let resolve_target =
+            |target: &str| self.resolve_type(target, current_namespace, shape_to_namespace);
+
+        match shape {
+            crate::types::SmithyShape::Structure {
+                members,
+                documentation,
+                traits,
+            } => {
+                let mut def = String::new();
+
+                if let Some(doc) = documentation {
+                    def.push_str(&format!("/// {}\n", doc));
+                }
+
+                for smithy_trait in traits {
+                    def.push_str(&format!("@{}\n", self.trait_to_string(smithy_trait)));
+                }
+
+                def.push_str(&format!("structure {} {{\n", name));
+
+                for (member_name, member) in members {
+                    if let Some(doc) = &member.documentation {
+                        def.push_str(&format!("    /// {}\n", doc));
+                    }
+
+                    for smithy_trait in &member.traits {
+                        def.push_str(&format!("    @{}\n", self.trait_to_string(smithy_trait)));
+                    }
+
+                    let resolved_target = resolve_target(&member.target);
+                    def.push_str(&format!("    {}: {}\n", member_name, resolved_target));
+                }
+
+                def.push('}');
+                def
+            }
+            crate::types::SmithyShape::Union {
```

Comment:
```suggestion
            types::SmithyShape::Union {
```
import and use it, fix in other places too